Twig/GlobalVariables.php

```diff
@@ -15,31 +15,52 @@
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
- * Class GlobalVariables.
- *
  * @author  Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
 class GlobalVariables
 {
     /**
      * @var ContainerInterface
+     *
+     * @deprecated Since version 3.4, will be removed in 4.0.
+     * NEXT_MAJOR : remove this property
      */
     protected $container;
 
     /**
-     * @param ContainerInterface $container
+     * @var Pool
      */
-    public function __construct(ContainerInterface $container)
+    protected $adminPool;
+
+    /**
+     * @param ContainerInterface|Pool $adminPool
+     */
+    public function __construct($adminPool)
     {
-        $this->container = $container;
+        // NEXT_MAJOR : remove this block and set adminPool from parameter.
+        if ($adminPool instanceof ContainerInterface) {
+            @trigger_error(
+                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
+                version 3.4 and will be removed in 4.0. Use Sonata\AdminBundle\Admin\Pool instead.',
```

Comment:
You can't know if `3.4` is correct until you actually merge, so `3.x` is safer : what if we release `3.5` and merge this by mistake?